src/foam/nanos/crunch/PreventMultipleSpidDAO.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch',
+  name: 'PreventMultipleSpidDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: 'DAO decorator that prevents assigning the user more than one spid',
+
+  javaImports: [
+    'foam.dao.DAO',
+    'foam.mlang.predicate.AbstractPredicate',
+    'foam.nanos.auth.ServiceProvider',
+    'foam.nanos.auth.ServiceProvider',
+    'static foam.mlang.MLang.*'
+  ],
+
+  methods: [
+    {
+      name: 'put_',
+      javaCode: `
+        UserCapabilityJunction ucj = (UserCapabilityJunction) obj;
+        Capability targetCapability = (Capability) ucj.findTargetId(x);
+
+        if ( targetCapability == null || ! ( targetCapability instanceof ServiceProvider ) ) super.put_(x, obj);
+
+        ServiceProvider sp = (ServiceProvider) targetCapability;
+
+        AbstractPredicate serviceProviderTargetPredicate = new AbstractPredicate(x) {
+          @Override
+          public boolean f(Object obj) {
+            UserCapabilityJunction ucj = (UserCapabilityJunction) obj;
+            Capability c = (Capability) ucj.findTargetId(x);
+            return c instanceof ServiceProvider;
+          }
+        };
```

Comment:
just an idea, if we use a ucj subtype specific for service provider eg. UserServiceProviderJunction then the predicate would just be `INSTANCE_OF(UserServiceProviderJunction.class)` thus avoiding the need to go through association. @jlhughes @kgrgreer what do you think?

Reply:
@lchanmann I'm not complete familiar with 'association', but the it 'sounds' like the subtype would make the logic cleaner. 
BUT - this should be happening infrequently, as just discussed, elsewhere is an error condition, so perhaps overkill for the odd use.